src/adapters/kafka/index.ts

```diff
@@ -65,133 +41,22 @@ class KafkaAdapter extends Adapter {
 
-      // this.kafka = consumer.connect({
-      //   host: url.host,
-      //   port: url.port || (url.protocol === 'kafka:' ? 1883 : 8883),
-      //   protocol: url.protocol.substr(0, url.protocol.length - 1),
-      //   clientId: serverBinding && serverBinding.clientId,
-      //   clean: serverBinding && serverBinding.cleanSession,
-      //   will: serverBinding && serverBinding.will && {
-      //     topic: serverBinding && serverBinding.lastWill && serverBinding.lastWill.topic ? serverBinding.lastWill.topic : undefined,
-      //     qos: serverBinding && serverBinding.lastWill && serverBinding.lastWill.qos ? serverBinding.lastWill.qos : undefined,
-      //     payload: serverBinding && serverBinding.lastWill && serverBinding.lastWill.message ? serverBinding.lastWill.message : undefined,
-      //     retain: serverBinding && serverBinding.lastWill && serverBinding.lastWill.retain ? serverBinding.lastWill.retain : undefined,
-      //   },
-      //   keepalive: serverBinding && serverBinding.keepAlive,
-      //   username: userAndPasswordSecurityReq ? process.env.GLEE_USERNAME : undefined,
-      //   password: userAndPasswordSecurityReq ? process.env.GLEE_PASSWORD : undefined,
-      //   ca: X509SecurityReq ? certs : undefined,
-      // })
-
-      this.kafka.on('connect', () => {
-        if (Array.isArray(subscribedChannels)) {
-          subscribedChannels.forEach((topic) => {
-            const operation = this.parsedAsyncAPI.channel(channel).publish()
-            const binding = operation.binding('kafka')
-            this.kafka.subscribe(channel, {
-              qos: binding && binding.qos ? binding.qos : 0,
-            })
-          })
-        }
+    const run = async () => {
+      await consumer.connect() 
+      const subscribedChannels = this.getSubscribedChannels()
+      await consumer.subscribe({ topics: subscribedChannels, fromBeginning: true })
+      await consumer.run({
+        eachMessage: async ({ topic, partition, message }) => {
+          const prefix = `${topic}[${partition} | ${message.offset}] / ${message.timestamp}`
+          console.log(`- ${prefix} ${message.key}#${message.value}`)
+        },
       })
+    }
 
-      // this.kafka.on('message', (topic, partition, message) => {
-      //   const msg = this._createMessage(message as IPublishPacket)
-      //   this.emit('message', msg, this.kafka)
-      // })
-
-      this.kafka.on('reconnect', () => {
-        this.emit('reconnect', {
-          connection: this.kafka,
-          channels: this.channelNames,
-        })
-      })
-      
-      this.kafka.on('close', () => {
-        this.emit('close', {
-          connection: this.kafka,
-          channels: this.channelNames,
-        })
-      })
-
-      this.kafka.on('error', (error) => {
-        this.emit('error', error)
-      })
+    run().catch(console.error)
+    await consumer.disconnect()
```

Comment:
If you call `disconnect`, it's going to only consume one message and then stop. We shouldn't ever disconnect from the broker.